tools/analysis_tools/analyze_results.py

```diff
@@ -341,22 +384,33 @@ def main():
 
     if args.cfg_options is not None:
         cfg.merge_from_dict(args.cfg_options)
-    cfg.data.test.test_mode = True
-
-    cfg.data.test.pop('samples_per_gpu', 0)
-    if cfg.data.train.type in ('MultiImageMixDataset', 'ClassBalancedDataset',
-                               'RepeatDataset', 'ConcatDataset'):
-        cfg.data.test.pipeline = get_loading_pipeline(
-            cfg.data.train.dataset.pipeline)
+    cfg.test_dataloader.dataset.test_mode = True
+
+    cfg.test_dataloader.pop('batch_size', 0)
+    if cfg.train_dataloader.dataset.type in ('MultiImageMixDataset',
+                                             'ClassBalancedDataset',
+                                             'RepeatDataset', 'ConcatDataset'):
+        cfg.test_dataloader.dataset.dataset.pipeline = get_loading_pipeline(
```

Comment:
Why `cfg.test_dataloader.dataset.dataset.pipeline` ?

test dataset will not be a dataset like 'MultiImageMixDataset',                                       'ClassBalancedDataset',  RepeatDataset', 'ConcatDataset'